tools/pyright_tool.py

```diff
@@ -46,18 +46,24 @@
 import tempfile
 from dataclasses import dataclass
 from pathlib import Path
-from typing import TYPE_CHECKING, Any, Final, Iterator, TypedDict
+from typing import Dict  # for python38 support, must be used in type aliases
+from typing import List  # for python38 support, must be used in type aliases
+from typing import TYPE_CHECKING, Any, Iterator
+from typing_extensions import (  # for python37 support, is not present in typing there
+    Final,
+    TypedDict,
+)
 
 import click
 
 if TYPE_CHECKING:
-    LineIgnores = list["LineIgnore"]
+    LineIgnores = List["LineIgnore"]
```

Comment:
type aliases hidden under `if TYPE_CHECKING` don't execute, so this should not be a problem?

Reply:
It should not be during runtime, but for the sake of the `pyright` tests, we need to also change it here, otherwise, it would complain about violation here